test/e2e/set/nodeconfig/nodeconfig_disksetup.go

```diff
@@ -283,23 +241,34 @@ var _ = g.Describe("Node Setup", framework.Serial, func() {
 		g.Entry("out of three loop devices", 3),
 	)
 
-	g.It("should propagate degraded conditions for invalid configuration", func() {
+	type degradedEntry struct {
+		nodeConfigFunc            func() *scyllav1alpha1.NodeConfig
+		preNodeConfigCreationFunc func(ctx context.Context, nc *scyllav1alpha1.NodeConfig) func(context.Context)
+	}
+
+	g.DescribeTable("should propagate mount controller's degraded condition", func(e *degradedEntry) {
 		ctx, cancel := context.WithTimeout(context.Background(), testTimeout)
 		defer cancel()
 
-		nc := ncTemplate.DeepCopy()
+		nc := e.nodeConfigFunc()
 
-		nc.Spec.LocalDiskSetup = &scyllav1alpha1.LocalDiskSetup{
-			Mounts: []scyllav1alpha1.MountConfiguration{
-				{
-					Device:             fmt.Sprintf("/dev/%s", f.Namespace()),
-					MountPoint:         fmt.Sprintf("/mnt/%s", f.Namespace()),
-					FSType:             string(scyllav1alpha1.XFSFilesystem),
-					UnsupportedOptions: []string{"prjquota"},
-				},
-			},
+		if e.preNodeConfigCreationFunc != nil {
+			cleanupFunc := e.preNodeConfigCreationFunc(ctx, nc)
+			defer cleanupFunc(ctx)
```

Comment:
use g.DeferCleanup. Context may be already cancelled during defer (if timeout happens) so it should use context.Background with optional additional timeout.

Reply:
good point, fixed, thanks

Reply:
DeferCleanup called in an `It` registeres a node behaving like AfterEach. Given our framework calls cleanup methods and removes the client in an AfterEach node, and AfterEach nodes have not guaranteed order, this could result in the `cleanupFunc` being called after the namespace and client are cleared. Hence using DeferCleanup would require significant changes to how our framework and all the tests are set up. 
I gave it a spec context instead, which shouldn't be invalidated when the test times out (since the timeout is not configured on the node but inside it).